src/ContinuityAdvElemKernel.C

```diff
@@ -0,0 +1,160 @@
+  dataPreReqs.add_master_element_call(SCS_AREAV);
+
+  // manage dndx
+  if ( !shiftPoisson_ || !reducedSensitivities_ )
+    dataPreReqs.add_master_element_call(SCS_GRAD_OP);
+  if ( shiftPoisson_ || reducedSensitivities_ )
+    dataPreReqs.add_master_element_call(SCS_SHIFTED_GRAD_OP);
+}
+
+template<typename AlgTraits>
+ContinuityAdvElemKernel<AlgTraits>::~ContinuityAdvElemKernel()
+{}
+
+template<typename AlgTraits>
+void
+ContinuityAdvElemKernel<AlgTraits>::setup(const TimeIntegrator& timeIntegrator)
+{
+  const double dt = timeIntegrator.get_time_step();
+  const double gamma1 = timeIntegrator.get_gamma1();
+  projTimeScale_ = dt / gamma1;
+}
+
+template<typename AlgTraits>
+void
+ContinuityAdvElemKernel<AlgTraits>::execute(
+  SharedMemView<double **>& lhs,
+  SharedMemView<double *>& rhs,
+  stk::mesh::Entity /*element*/,
+  ScratchViews& scratchViews)
+{
+  SharedMemView<double*>& v_densityNp1 = scratchViews.get_scratch_view_1D(*densityNp1_);
+  SharedMemView<double*>& v_pressure = scratchViews.get_scratch_view_1D(*pressure_);
+
+  SharedMemView<double**>& v_velocity = scratchViews.get_scratch_view_2D(*velocityRTM_);
+  SharedMemView<double**>& v_Gpdx = scratchViews.get_scratch_view_2D(*Gpdx_);
+
+  SharedMemView<double**>& v_scs_areav = scratchViews.scs_areav;
+
+  SharedMemView<double***>& v_dndx = shiftPoisson_ ?
+    scratchViews.dndx_shifted : scratchViews.dndx;
+  SharedMemView<double***>& v_dndx_lhs = (!shiftPoisson_ && reducedSensitivities_)?
+    scratchViews.dndx_shifted : scratchViews.dndx;
+
+  for (int ip = 0; ip < AlgTraits::numScsIp_; ++ip) {
+    const int il = lrscv_[2*ip];
+    const int ir = lrscv_[2*ip+1];
+
+    double rhoIp = 0.0;
+    for (int j = 0; j < AlgTraits::nDim_; ++j) {
+      v_uIp_(j) = 0.0;
+      v_rho_uIp_(j) = 0.0;
+      v_Gpdx_Ip_(j) = 0.0;
+      v_dpdxIp_(j) = 0.0;
+    }
+
+    for (int ic = 0; ic < AlgTraits::nodesPerElement_; ++ic) {
+      const double r = v_shape_function_(ip, ic);
+      const double nodalPressure = v_pressure(ic);
+      const double nodalRho = v_densityNp1(ic);
+
+      rhoIp += r * nodalRho;
+
+      double lhsfac = 0.0;
+      for (int j = 0; j < AlgTraits::nDim_; ++j) {
+        v_Gpdx_Ip_(j) += r * v_Gpdx(ic, j);
+        v_uIp_(j)     += r * v_velocity(ic, j);
+        v_rho_uIp_(j) += r * nodalRho * v_velocity(ic, j);
+        v_dpdxIp_(j)  += v_dndx(ip, ic, j) * nodalPressure;
+        lhsfac += -v_dndx_lhs(ip, ic, j) * v_scs_areav(ip, j);
+      }
+
+      lhs(il,ic) += lhsfac;
+      lhs(ir,ic) -= lhsfac;
+    }
+
+    // assemble mdot
+    double mdot = 0.0;
+    for (int j = 0; j < AlgTraits::nDim_; ++j) {
+      mdot += (interpTogether_ * v_rho_uIp_(j) + om_interpTogether_ * rhoIp * v_uIp_(j) -
+               projTimeScale_ * ( v_dpdxIp_(j) - v_Gpdx_Ip_(j))) * v_scs_areav(ip,j);
+    }
+
+    // residuals
+    rhs(il) -= mdot / projTimeScale_;
+    rhs(ir) += mdot / projTimeScale_;
+  }
+}
+
+INSTANTIATE_SUPPLEMENTAL_ALGORITHM(ContinuityAdvElemKernel);
```

Comment:
Good. I thought we removed most of those. In general, only the ADV and DIFF flavors of the SuppAlgs should be removed since the interface to plug in SuppAlgs is alone through specialized Cont, Scalar and Momentum algorithms which already include ADV and DIFF.